payjoin/src/receive/v1/mod.rs

```diff
@@ -1031,7 +1027,7 @@ pub(crate) mod test {
         let script = Script::new();
         let replace_receiver_script_pubkey = wants_outputs.substitute_receiver_script(script);
         match replace_receiver_script_pubkey {
-            Ok(_) => panic!("Expected error, got success"),
+            Ok(_) => panic!("Payjoin receiver script pubkey has been modified and should error"),
```

Comment:
You didn't write the match, but it can be simplified to one assert_eq without a match, again.

```rs
assert_eq!(
            replace_receiver_script_pubkey.unwrap_err(),
            OutputSubstitutionError::from(
                InternalOutputSubstitutionError::ScriptPubKeyChangedWhenDisabled
            ),
            "Payjoin receiver script pubkey has been modified and should error"
        );
```